File: abc/abc145/abc145_c.cs
// https://atcoder.jp/contests/abc145/submissions/8488483

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());

            var town = new int[N][];
            for (int i = 0; i < N; i++)
            {
                town[i] = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            }

            var dist = 0.0;
            var x = 0;
            var y = 0;
            for (int i = 0; i < N; i++)
            {
                for (int j = 0; j < N; j++)
                {
                    if (i == j)
                    {
                        continue;
                    }
                    x = Math.Abs(town[i][0] - town[j][0]) * Math.Abs(town[i][0] - town[j][0]);
                    y = Math.Abs(town[i][1] - town[j][1]) * Math.Abs(town[i][1] - town[j][1]);
                    dist += Factorial(N - 1) * Math.Sqrt(x + y);
                }
            }

            Console.WriteLine(dist / Factorial(N));
        }

        private static int Factorial(int n)
        {
            var result = 1;
            for (int i = 0; i < n; i++)
            {
                result *= (i + 1);
            }
            return result;
        }
    }
}

File: abc/abc145/abc145_c.fs
// https://atcoder.jp/contests/abc145/submissions/12155167

let N = stdin.ReadLine() |> int

let xy =
    [| for _ in [ 1..N ] do
        yield (stdin.ReadLine().Split() |> Array.map int) |]

let rec fact r n =
    if n = 1 then r
    else fact (r * n) (n - 1)

let rec resolver r i j =
    if i = N then r / float (fact 1 N)
    else
        let rec loop r j =
            if j = N then resolver r (i + 1) 0
            elif i = j then loop r (j + 1)
            else
                let x = abs (xy.[i].[0] - xy.[j].[0]) * abs (xy.[i].[0] - xy.[j].[0])
                let y = abs (xy.[i].[1] - xy.[j].[1]) * abs (xy.[i].[1] - xy.[j].[1])
                loop (r + float (fact 1 (N - 1)) * sqrt (float (x + y))) (j + 1)
        loop r j

resolver 0.0 0 0 |> stdout.WriteLine
